scripts/bash_includes/tc/lua

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: BSD-2-Clause
+
+function download_lua {
+   local lua=$1
+   local version=$2
+   local filename=${lua}
+   local tarname="${lua}-${version}.tar.gz"
+   local url="https://www.lua.org/ftp/"
+   
+   download_file_in_cache "${url}" "${tarname}" lua
+   extract_cachefile_tar_gz "${tarname}" "${lua}-${version}" "${lua}"
+}
+
+all_funcs_list+=(build_lua)
+function build_lua {
+
+   pushd ${ARCH} 
+
+   lua="lua"
+   lua_version="5.4.1"
+   reset_cc_vars
+
+   show_work_on_component_msg "LUA"
+
+   if ! [ -d "${lua}" ]; then
```

Comment:
The idea across all the build toolchain scripts is that they can be run over and over again, without effect except for the first time. They don't support rebuild (except when we manually delete the $ARCH/<app> directory. So, the skeleton for the build_XXX funcs is something like:

```
all_funcs_list+=(build_XXX)
function build_XXX {

   pushd ${ARCH}

   if ! [ -d XXX ]; then

      show_work_on_component_msg "XXX"
      reset_cc_vars

      download_XXX
      cd XXX

      if [[ "$HOST_ARCH" != "$BUILD_ARCH" ]]; then

         # Case HOST_ARCH == ARCH
         # Note: if HOST_ARCH not in [BUILD_ARCH, ARCH], we fail earlier.
         #
         # Note: NOT TESTED yet.
         # Remember: this scenario (building on $ARCH) is _not_ officially
         # supported.

         export SYS_CC="$CC"
         export SYS_CXX="$CXX"
         export CC_POST_FLAGS="-specs $MUSL_INSTALL/lib/musl-gcc.specs"

      else
         # Default case
         set_cc_vars_to_tc
      fi

      #
      # INSERT APP SPECIFIC CONFIG FLAGS HERE
      #
      run_command2 "make $flags" build.log

   else

      show_skip_component_msg "XXX"
   fi

   popd
}
```

Reply:
I have changed the function structure as per the standard skeleton.